vortex-array/src/array/primitive/mod.rs

```diff
@@ -134,6 +141,65 @@ impl<'a> PrimitiveArray<'a> {
     pub fn into_buffer(self) -> Buffer<'a> {
         self.into_array().into_buffer().unwrap()
     }
+
+    fn subtract_scalar_integer<
```

Comment:
Why is this inside the PrimitiveArray impl? Just make it a function inside primiive/compute/